src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -218,7 +218,8 @@ public int CurrentWorkspaceIndex
 
                     // Keep DynamoModel.CurrentWorkspace update-to-date
                     int modelIndex = model.Workspaces.IndexOf(currentWorkspaceViewModel.Model);
-                    this.ExecuteCommand(new DynamoModel.SwitchTabCommand(modelIndex));
+                    ExecuteCommand(new DynamoModel.SwitchTabCommand(modelIndex));
+                    (HomeSpaceViewModel as HomeWorkspaceViewModel).UpdateRunStatusMsgBasedOnStates();
```

Comment:
If (HomeSpaceViewModel as HomeWorkspaceViewModel) is null, does it will crash? if is true I think you should add a validation.

Reply:
Missed this comment but yes, HomeSpaceViewModel is always `HomeWorkspaceViewModel`